pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -455,7 +455,7 @@ class ModelInterpreter implements Serializable {
                     "${k}=${v.call()}"
                 }catch (NullPointerException e){
 
-                    throw new IllegalArgumentException(format("Invalid var declared in environment: %s", k))
+                    throw new IllegalArgumentException(format("Invalid variable (%s) declared in environment; There are some problems with its value", k))
```

Comment:
So in practice, this is only going to show up if there's a reference to a non-existent variable in the value, right? Hence the `NullPointerException` - so in that case, we could probably be more specific and say something like `One or more variables in its value do not exist`?

Also, this should go in `Messages.properties` so that it can be localized.